tools/lib/uobjnew.py

```diff
@@ -98,7 +98,20 @@
     u64 classptr = 0, size = 0;
     bpf_usdt_readarg(2, ctx, &classptr);
     bpf_usdt_readarg(4, ctx, &size);
-    bpf_probe_read_user(&key.name, sizeof(key.name), (void *)classptr);
+    // The reason why read the third arg twice with different types here:
+    // According to the source code of OpenJDK:
+    // the declaration type of the third arg is uintptr_t(see hotspot.d)
+    // but the actual type is int(see SharedRuntime::dtrace_object_alloc)
```

Comment:
If the actual type is int, should we always read len32?

Reply:
At present, it should only be int, but I am not sure whether the OpenJDK community will fix this problem in the future.
If it will be fixed, we are currently not sure whether it is int or uintptr_t at the end.
So my current implementation is for better compatibility considerations.